docs/contributing/commit-signing.mdx

```diff
@@ -16,61 +16,144 @@ For example, on 3 August 2022, Stephen Lacy
+latest security best practices </strong> (see the
+[PGP problem](https://www.latacora.com/blog/2019/07/16/the-pgp-problem/) for more details).
 
 :::
 
-### Configuration
+### Standard SSH Key Configuration
 
-1. Configure git to sign using SSH.
+1. Ensure you have globally configured your git email address:
+
+   ```bash
+   git config --global user.email "you@example.com"
+   ```
+
+2. Generate an SSH key for commit signing:
+
+   ```bash
+   # path to save your private signing key
+   KEY_FILE=~/.ssh/bw-signing
+   ssh-keygen -f $KEY_FILE -C "$(git config --global --get user.email)" -t ed25519
+   ```
+
+   :::tip
+
+   Remember to protect the key with a [strong passphrase or password][password-generation].
+   Alternatively set up your [yubikey as your signing key](#hardware-backed-ssh-key-configuration).
+
+   :::
+
+3. Configure git to sign using SSH.
 
    ```bash
    git config --global gpg.format ssh
    git config --global user.signingkey "${KEY_FILE}.pub"
    ```
 
-2. Follow the [Github documentation][github-verification] to configure commit signing
+4. Follow the
+   [SSH GitHub documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification)
+   to configure commit signing
 
-3. Configure your preferred git tool below
+5. Configure your preferred git tool below ([CLI](#commit-signing-with-git-cli),
+   [VS Code](#commit-signing-with-visual-studio-code),
+   [SourceTree](#commit-signing-with-sourcetree))
 
-4. Push a test commit to Github and ensure that the "Verified" badge appears next to the commit
+6. Push a test commit to GitHub and ensure that the "Verified" badge appears next to the commit
    description:
 
-   ![Image showing the Verified badge in Github](./commit-signing.png)
+   ![Image showing the Verified badge in GitHub](./commit-signing.png)
 
-### Command Line
+7. (Optional) See Avoiding SSH Key Passphrase Prompts Per Commit below to avoid being prompted for
+   your SSH key passphrase every time you sign a commit.
 
-- After configuring commit signing, you can sign a commit by using the `-S` flag:
+### Avoiding SSH Key Passphrase Prompts Per Commit
 
-  ```bash
-  git commit -S
-  ```
+#### macOS
 
-- To avoid using the `-S` flag every time, you can sign all commits and tags by default:
+On macOS, you can use the built-in Keychain to store your SSH key passphrase so you don't have to
+enter it every time you sign a commit.
 
-  ```bash
-  git config --global commit.gpgSign true
-  git config --global tag.gpgSign true
-  ```
+1.  Add your SSH key to the ssh-agent and store your passphrase in the Keychain:
+
+    ```bash
+    ssh-add --apple-use-keychain ~/.ssh/bw-signing
+    ssh-add -l # verify the key was added
+    ```
+
+2.  The above command will need to be ran each time you restart your computer. If you would like to
+    automate this process, then you can leverage LaunchAgents to run the command on startup.
```

Comment:
Perhaps add an external link to LaunchAgents documentation?

Reply:
cc7a04b9a34ee239bba8ce2e23ef7754f790148d